ctest/src/lib.rs

```diff
@@ -851,25 +853,54 @@ impl TestGenerator {
             cfg.include(p);
         }
 
-        let stem = out.file_stem().unwrap().to_str().unwrap();
-        cfg.target(&target)
-            .out_dir(out.parent().unwrap())
-            .compile(&format!("lib{stem}.a"));
+        let stem = out
+            .file_stem()
+            .ok_or_else(|| Error::from("Output file has no stem"))?
+            .to_str()
+            .ok_or_else(|| Error::from("Output filename is not valid UTF-8"))?;
+
+        let parent = out
+            .parent()
+            .ok_or_else(|| Error::from("Output file has no parent directory"))?;
+
+        cfg.target(&target).out_dir(parent);
+
+        match catch_unwind(|| cfg.compile(&format!("lib{stem}.a"))) {
```

Comment:
Unfortunately not quite :) we want to change all internal functions to return a `Result` as well so it's easier to know what might fail and for simpler testing.

It should be pretty easy to find+replace any `.unwrap()` with `?`, then update the function signatures anywhere it complains.